src/ethereum/EthereumEngine.ts

```diff
@@ -739,13 +744,34 @@ export class EthereumEngine extends CurrencyEngine<
   // Public methods
   // ****************************************************************************
 
-  async startEngine(): Promise<void> {
-    this.currencyEngineCallbacks.onSubscribeAddresses([
-      {
-        address: this.walletLocalData.publicKey,
-        checkpoint: this.walletLocalData.highestTxBlockHeight.toString()
-      }
-    ])
+  async startEngine(opts: EdgeStartEngineOptions = {}): Promise<void> {
+    const realAddress = this.walletLocalData.publicKey
+
+    // Sync in-memory decoy addresses with what core has saved
+    if (opts.subscribedAddresses != null) {
+      this.subscribedAddresses = opts.subscribedAddresses
+    }
+
+    // Ensure that the real address is in the list
+    this.subscribedAddresses.unshift({
```

Comment:
If the list is > 0 length, how could you not have the main address in the list? Why not just add the real address if the list is empty and never again.

Reply:
`this.walletLocalData.publicKey` is not available at construction-time for the EthereumEngine which is were we initialize `this.subscribedAddresses`. As an alternative, we could just initialize the `this.subscribeAddresses` here if `.length === 0`.